src/hotspot/share/nmt/nMemLimit.hpp

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2023 SAP SE. All rights reserved.
+ * Copyright (c) 2023, 2024, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2025 Amazon.com Inc. or its affiliates. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_NMT_NMEMLIMIT_HPP
+#define SHARE_NMT_NMEMLIMIT_HPP
+
+#include "memory/allStatic.hpp"
+#include "nmt/memTag.hpp"
+#include "utilities/debug.hpp"
+#include "utilities/globalDefinitions.hpp"
+
+enum class NMemType : int {
+  Malloc,
+  Mmap
+};
+
+enum class NMemLimitMode {
+  trigger_fatal = 0,
+  trigger_oom   = 1
+};
+
+struct nMemlimit {
+  size_t sz;            // Limit size
+  NMemLimitMode mode; // Behavior flags
+};
+
+// forward declaration
+class outputStream;
```

Comment:
Style: Remove the `// forward declaration` comment, and move it up right after the includes.